### PR TITLE
Add release process

### PR DESCRIPTION
**Pull request by SergioGasquez**
Automates creating the binary artifacts and publishing the package to crates.io.  The release process is basically copied from `espup`.

We need to build the package before publishing as it needs to create the `template_files.rs`, and for this same reason, we need `--allow-dirty`

- [ ] We need to add the `${{ secrets.CARGO_API_KEY }}` secret

Test release in my fork https://github.com/SergioGasquez/esp-generate/actions/runs/11891708349

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  publish-artifacts:
+    name: Generating artifacts for ${{ matrix.job.target }}
+    runs-on: ${{ matrix.job.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - os: macos-latest
+            target: x86_64-apple-darwin
+          - os: ubuntu-20.04
```
Comment (jessebraham) [inline .github/workflows/cd.yml:21]: Is there any reason to use `20.04` instead of `latest`? Probably doesn't matter but just curious

Comment (jessebraham) [reply .github/workflows/cd.yml:21]: Ahh okay that rings a bell haha, thanks

**Review by jessebraham — APPROVED** [review]

**Review by MabezDev — APPROVED** [review]

**Review by MabezDev — CHANGES_REQUESTED** [review]
Can you roll back that last commit, sorry!

**Review by MabezDev — APPROVED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (SergioGasquez) [reply .github/workflows/cd.yml:21]: As I did nor recall this, I was just gonna try with ubuntu latest

**Pull request merged**